### ISSUE TITLE
How to remove uploaded artifact?

### ISSUE THREAD

**Issue by williamli** [actions/upload-artifact#5]
It is possible to remove an uploaded artifact?

**Comment by 1138-4EB** [comment]
*ping*...

**Comment by TingluoHuang** [comment]
@madhurig for artifact questions

**Comment by madhurig** [comment]
Currently there is no experience to remove uploaded artifacts. Is there a scenario you have that you would need this feature for or are you looking to remove an artifact accidentally uploaded?

**Comment by 1138-4EB** [comment]
> Is there a scenario you have that you would need this feature for or are you looking to remove an artifact accidentally uploaded?

We expect to use GitHub Action and artifacts to provided 'nightly builds' without having to deal with GitHub Releases. I.e., we currently build a tarball and run tests with it, as a regular CI job. We expect to reuse the same tarball, instead of trigger all the release procedure (which is meant for tagged commits/releases).

In the situations when a tarball passes the tests but it is not valid for any other reason, we want to remove the artifacts. This would allow us to tell users: *you can pick the latest successful (green) run which contains artifacts*, and ensure that the first file they download will work.

BTW, can you please have a look at #3?

**Comment by williamli** [comment]
> Is there a scenario you have that you would need this feature for or are you looking to remove an artifact accidentally uploaded?

@madhurig I was using artifacts to pass docker images and some gcloud configs between jobs so that I don't have to re-auth with the gcloud sdk in different jobs. It might not be the most optimised way but I was exploring that idea a few weeks ago. It will be good if we can remove these artifacts at the end of the jobs to clean up.

**Comment by madhurig** [comment]
@williamli - Thanks for elaborating. We will evaluate this feature.

**Comment by arnavzoman** [comment]
We need this feature because there is a 1GB limits on artifacts and we do not need the older ones.

**Comment by mieszko4** [comment]
Is there some API to remove uploaded artifacts instead?
I would like to set up a github action cronjob to remove artifacts older than 1 month.

**Comment by kf6kjg** [comment]
As far as I can tell there’s not only no API, but no manual removal route either. This is because, as noted above, they’d not been evaluated the needed user experience. For instance, while they could release an api, would an expiration policy work better? Or maybe something else? Perhaps a totally different tool is needed, eg a cleaner way to pass messages or files from one job to another. 
Hence they are on the discovery phase for this and need our input on how and why we are using artifacts.

ATM my use case is that of inter job communications.  Later I might use artifacts to hold build or test results for review.

**Comment by mieszko4** [comment]
Thanx @kf6kjg 
For me I guess expiration policy would work, too.

I mean, not having a way to remove artifacts and at the same time billing for over-filled storage is a contradiction to me.

Btw my usecase: we integrated cypress with videos stored as artifacts - there is no need to have them for longer than few days.

**Comment by mscoutermarsh** [comment]
Hi all, we're working on an API for deleting an artifact. Thanks for sharing your use cases ❤️.

I don't have a timeline yet that I can share on when it will be available.

**Comment by hellwolf** [comment]
There is a workaround now, you could create an empty directory and upload a new artifact of the same name with path of the empty directory.

```
      - name: create empty directory
        run: |
          mkdir empty.d

      - name: empty dist archive
        uses: actions/upload-artifact@v1
        with:
          name: dist
          path: empty.d
```

**Comment by kf6kjg** [comment]
Unless there's special server code for that case, a zip file of an empty directory still occupies ~80kB of space. Certainly less than the zip it replaced, but continual growth is still growth even if you manage to reduce the rate.
All that said, while I don't consider it a workaround, @hellwolf you've got a clever technique to reduce the impact.  Thanks! :D

**Comment by dodie** [comment]
I could not manage to make @hellwolf 's workaround to work:

After uploading the empty folder with the same name as an artifact I could download the original artifact from action's UI, and the size displayed matched with the original file size.

When I added an empty file to `empty.d` with the name of the original file, I could no longer access the original file as if it was replaced by the empty one, but the UI still displayed the original file size.

Even though I could not access the artifact in the last case, I'm not sure it's actually replaced with the empty file behind the scenes.

**Comment by hellwolf** [comment]
@dodie yes, it seems github changed some logic, the workround is no longer working!!

**Comment by dentarg** [comment]
Because of this perhaps? https://github.blog/changelog/2020-01-14-github-actions-changes-to-artifact-download-experience/

**Comment by gimenete** [comment]
Actions engineer here 👋  There's now the possibility to delete the artifact in the UI. You'll see a trash can icon if you have permissions to write in the repository contents.

<img width="336" alt="Screen Shot 2020-01-07 at 16 08 06" src="https://user-images.githubusercontent.com/50486/72544679-ca8f8380-3855-11ea-8542-4fccf55faa80.png">

**Comment by xt0rted** [comment]
@gimenete not sure if this is intentional or not but there's no confirmation dialog on mobile like there is on desktop.

**Comment by konradpabjan** [comment]
@dodie @hellwolf @dentarg  The workaround should no longer work, we recently made some backend changes to our artifacts service. The recent changes to the download experience were part of a broader backend update but not directly responsible for the workaroud no longer working.

Recommend to use the new UI now 😀 That will completely delete the artifact and there won't be any empty directories.

I've updated the `README` in this repo with some information about deleting an artifact from the UI with a new screenshot. A DELETE API for artifacts is in development so there will be more ways to delete an artifact down the road. Going to close this issue now that there is at least one way to delete an artifact.

**Comment by JCMais** [comment]
is it possible to remove the artifacts from an action / api?

**Comment by eine** [comment]
> is it possible to remove the artifacts from an action / api?

> A DELETE API for artifacts is in development so there will be more ways to delete an artifact down the road.

**Comment by chrispat** [comment]
https://developer.github.com/v3/actions/artifacts/#delete-an-artifact

**Comment by kf6kjg** [comment]
And there was joyful singing! Note to get that incorporated into the actions that need it! :D

**Comment by engahmed1190** [comment]
> https://developer.github.com/v3/actions/artifacts/#delete-an-artifact

How we can use it in GitHub workflow .

**Comment by christian-korneck** [comment]
now that there's an api (thanks for that!) an official `actions/upload-artifact@v1` action would be awesome ;-)

**Comment by alonsosprey** [comment]
> now that there's an api (thanks for that!) an official `actions/upload-artifact@v1` action would be awesome ;-)

Is there any open ticket for this?

**Comment by up-kalel** [comment]
> > now that there's an api (thanks for that!) an official `actions/upload-artifact@v1` action would be awesome ;-)
> 
> Is there any open ticket for this?

I don't think so...

**Comment by kolpav** [comment]
If anyone wants to have simple expiration policy for their artifacts I wrote github action just for that. Its enough for my usecase if there is some functionality you would like to see let me know in issue. https://github.com/marketplace/actions/purge-artifacts

**Comment by craig-dae** [comment]
This just started killing our cicd. I tried @kolpav purge library, and I'm still getting the error. 

I'm building an app that takes about 5 minutes to build a 130mb, which it needs to upload as an artifact, so it can be downloaded by 3+ test servers to run parellel tests. We hit our artifact limit pretty quickly, and I don't know how to make space. 

I've manually deleted a bunch of artifacts from previous runs, and I ran @kolpav's action before doing my upload. I'm still getting the error. 

This is a blocker. Can anyone help?

**Comment by craig-dae** [comment]
I'm seeing https://developer.github.com/v3/actions/artifacts/#delete-an-artifact above. Anyone have an example of that being used in a github action?

**Comment by kolpav** [comment]
@craig-dae What do you see in https://github.com/settings/billing "Storage for Actions and Packages" section?

**Comment by craig-dae** [comment]
Nothing because I don't own the account. 

That's the other thing. GitHub can't expect the developer who is building the pipeline to have full access to the account, and insight onto that page. Generally, it's the CTO or someone one level below the CTO with that access. There needs to be a way for someone without billing-access to view the artifacts/actions information. I just found out yesterday that we have spent 72% of our GitHub actions for the month in the first two weeks of using them. 

Anyway, I think we're back in action. I think that @kolpav's action works, but I think there is a delay between clearing out artifacts, and GitHub letting you use the reclaimed space. About 2 hours after I ran the action, pipelines uploading artifacts started working again.

**Comment by christian-korneck** [comment]
@craig-dae yes, I’m also seeing a short delay between deleting artifacts and the usage/quota display in the account settings being updated.

**Comment by christian-korneck** [comment]
>> @craig-dae I'm seeing https://developer.github.com/v3/actions/artifacts/#delete-an-artifact above. Anyone have an example of that being used in a github action?

> @kolpav : If anyone wants to have simple expiration policy
thanks, that's useful!

I have another example for a different usecase (delete no longer needed artifacts at the end of the workflow run in which they have been created):
https://github.com/marketplace/actions/delete-run-artifacts

(I use it to delete temporary artifacts created by matrix build jobs after they have been uploaded to a github release and S3 bucket).

**Comment by thisismydesign** [comment]
Our team has created an Action to periodically clean up old artifacts:
https://github.com/c-hive/gha-remove-artifacts

It also supports keeping tagged (release) artifacts.

**Comment by variadico** [comment]
I'm using `upload-artifact` to share intermediary work between jobs. Idea taken from here: https://docs.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts

However, the intermediary work is junk on its own. It would be nice if I didn't have to manually go into the history and click delete all the time.

Maybe there should be a `actions/delete-artifact`. :thinking:

**Comment by omarabid** [comment]
The effect delay wasted me a good 30min as I was not sure if the action by @kolpav was not functioning or something else is going on...

**Comment by jeacott1** [comment]
I've discovered today that whilst deleting an artifact via the gui works, it doesn't seem to have any effect on the storage counts on the billing page. is this a glitch or working as expected?

**Comment by 96312** [comment]
@omarabid I used his workflow too. Ran it but the space doesn't seem to be freed even after running it.

**Comment by jdrydn** [comment]
> You've used 100% of included services for GitHub Storage (GitHub Actions and Packages).

Yeah, I've also ended up here after receiving a billing notification email from GitHub! Would love to see a GitHub-produced `actions/delete-artifact` to run after builds, but instead I'll settle for adding [a retention period](https://github.com/actions/upload-artifact#retention-period) to the artifacts as my use-case involves artifacts between jobs so persisting for a day is fine!

**Comment by JoseQuintas** [comment]
Imagine this situation:

An automatic workflow to create artifacts after each commit: 300MB of artifact
20 commits on each day x 90 days x 300MB = 540.000MB = 540GB
How to delete old runs of the same artifact after successfull creation?
This will reduce from 540GB to 300MB, and do not need to wait for 90 days.

At momment, may be I can create scripts to search what to delete.
Is there exists actions to delete specific workflow run or a list of workflow runs
or old runs of same workflow, when current workflow is sucessfull ?
From 540GB to 300MB, there exists a big difference, to user and to github.

Another option is to have repo artifacts, like release or pack.
Each workflow could replace the repo artifact, and not the workflow artifact.
Think about to reduce from 540GB to 300MB.

And we are talking about ONE workflow, think about all repositories on github, and how many space this can reduce at all.

Can reduce artifact life to explire after 1 day....
Ok, but if I do not make commit, I do not have artifact.
May be I am using artifact as release or pack, but why not?

Another option, could be have an option to an automatic-rerun after artifact expires.

I am using free account, not sure if 540GB is a problem, but I would like reduce this use.
For beginners like me, a ready to use action can help me to do more, and github do not lost space with trash.
1.000 repositories using 540GB, is 540TB. What about github have 540TB available?

**Comment by JoseQuintas** [comment]
Another possible solution: setup a size limit to a yml history.
And delete old artifacts when that size is reached.

Example: 
I build a artifact using mywork.yml of 100MB, and set limit to 500MB.
Only last 5 runs remains available, others will be deleted when a new run occurs.

**Comment by MohammedNoureldin** [comment]
So what is the situation, won't be any official `actions/delete-artifact` soon? For my use case (removing the build files remaining from **build job** after running a **dependant deploy job**) I plan to use this implementation:

https://github.com/GeekyEggo/delete-artifact

I have not tested it yet, but this seems to be exactly what I need.

**Comment by platformjb** [comment]
An alternative option for ephemeral files (e.g., using the output of one job in another) is to use S3. The AWS CLI is available in actions by default. S3 also supports lifecycle rules so objects can be automatically expired/deleted after a certain number of days (or you could do `aws s3 rm ...` at the end of the run.

Uploading artifacts can be done with:

```yaml
- name: Upload artifacts to S3
  run: |
    aws s3 cp artifacts.tar.gz s3://BUCKET_NAME/${GITHUB_RUN_NUMBER}/artifacts.tar.gz --no-progress
  env:
    AWS_ACCESS_KEY_ID: ${{ secrets.ARTIFACTS_AWS_ACCESS_KEY_ID }}
    AWS_SECRET_ACCESS_KEY: ${{ secrets.ARTIFACTS_AWS_SECRET_ACCESS_KEY }}
    AWS_DEFAULT_REGION: 'eu-west-1'
```

And downloading with...

```yaml
- name: Download artifacts from S3
  run: |
    aws s3 cp s3://BUCKET_NAME/${GITHUB_RUN_NUMBER}/artifacts.tar.gz artifacts.tar.gz
  env:
    AWS_ACCESS_KEY_ID: ${{ secrets.ARTIFACTS_AWS_ACCESS_KEY_ID }}
    AWS_SECRET_ACCESS_KEY: ${{ secrets.ARTIFACTS_AWS_SECRET_ACCESS_KEY }}
    AWS_DEFAULT_REGION: 'eu-west-1'
```

Using `GITHUB_RUN_NUMBER` keeps all artifact files under a common key for each run and avoids runs interfering with one another.

An IAM user would need to be configured with `PutObject` and `GetObject` permissions on the bucket. The access key information would then need to be added as repository secrets.

This has worked well for us with files ~100mb with no negative performance impact compared to GitHub artifacts (~4s to upload, ~8s to download). YMMV for larger files.

**Comment by kf6kjg** [comment]
@platformjb that's a clever use of tools and even can be extended to Azure as the Azure CLI is likewise included.

TL;DR: Just a warning - when uptime and/or consistent operation are critical features of your system, relying on secondary or tertiary external services isn't a good plan.

I'd be concerned about using a secondary service when doing this kind of task.  I've had bad experiences caused by anything from network outage between the two services.  For instance in a company I contract for they'd originally set up Azure WebApps to pull Docker container images from the GitHub container registry.  This worked well for over a year, until one day one of the the server instances crashed due to an internal bug and couldn't restart due to being unable to access GitHub's domain.  In this example the primary service was Azure, and it was implicitly depending on the availability of a secondary service, GitHub.  I resolved this by moving to the Azure Container Registry, as the likelihood of communications failures between one Azure service in the same region as another Azure service is very low.

On an unrelated note, I've had to rearchitect the build system to do as much as possible in a single job as I can, using GNU parallel when possible inside that.  The problem was that I was transmitting my entire build context, a gzipped tar of some 2-3 GB.  The system uploaded it once in the job that did a lot of common activities and checks, then downloaded it on every job that needed that prepared context, in lieu of actually cloning the repo.  This worked well, but I found that nearly 50% of the time was spent in compressing, uploading, downloading, and decompressing that file.  So I found a different architecture, inspired by several blog posts about different CICD systems, and now it's running much faster.

**Comment by jcansdale** [comment]
Here is a job you can include in your workflow that will automatically delete all artifacts created in previous workflow runs.

```yml
  delete-artifacts:
    runs-on: ubuntu-latest
    steps:
      - name: Delete artifacts from previous runs
        uses: actions/github-script@v3
        with:
          github-token: ${{ github.token }}
          script: |
            const runsResponse = await github.request("GET /repos/${{ github.repository }}/actions/runs")
            for (workflow_run of runsResponse.data.workflow_runs) {
              if (workflow_run.name == "${{ github.workflow }}" && workflow_run.status !== "in_progress") {
                console.log(`Deleting artifacts from workflow run ${workflow_run.id}`)
                const response = await github.request(`GET /repos/$/actions/runs/${workflow_run.id}/artifacts`)
                for (artifact of response.data.artifacts) {
                  console.log(`Deleting artifact with id ${artifact.id}`)
                  const deleteResponse = await github.request(`DELETE /repos/$/actions/artifacts/${artifact.id}`)
                }
              }
            }
```

**Comment by kolpav** [comment]
This issue being closed and after all these years there is still no way how to do something trivial as this is sad to be honest. You can't even set retention period to `0` you need to keep garbage after your runs at least for a day and UI doesn't really make it easier to manually delete artifacts.

**Comment by christian-korneck** [comment]
> This issue being closed and after all these years there is still no way how to do something so trivial is sad to be honest. 

It’s possible using an undocumented api request. See: https://github.com/GeekyEggo/delete-artifact

**Comment by kolpav** [comment]
@christian-korneck Thats not what I meant I see not keeping the artifacts as something that should be native GA feature all that needs to be done is to allow to do this `retention-days: 0` no undocumented apis no 20 version of delete artifact actions etc.

**Comment by christian-korneck** [comment]
@kolpav (Disclaimer: I'm not a maintainer here). I read it the way that this issue got closed because there's a way to delete artifacts both through the GUI and API. However, I believe there's no _documented_ way to delete an artifact *during the workflow run in which the artifact got created* (only after). Maybe open a new issue for that specific requirement?

**Comment by kolpav** [comment]
@christian-korneck Sorry to be so negative but I just gave up on GA its product I am forced to use and each time I need to touch it I get more frustrated with it not like I haven't tried my https://github.com/kolpav/purge-artifacts-action is being forked quite often maybe because its broken by now but serves as good starting point even at time of writing it I needed to do some hackery https://github.com/kolpav/purge-artifacts-action/blob/master/src/utils.ts#L8-L22 . Common theme with GA is to create bad API then propose the most complicated solution possible as fix instead of fixing the actual problem.

1. Lack of
```
on:
  pull_request:
    types: [merged]
```
Solution:
- `if: github.event.pull_request.merged == 'true'` but this still triggers the workflow!
- https://docs.github.com/en/github/administering-a-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule

2. Lack of
```
retention-days: 0
```
Solution: We know this story:)


3. Getting SHA of commit that triggered the workflow
This persons comment nicely summaries my frustration https://github.community/t/github-sha-not-the-same-as-the-triggering-commit/18286/4

These are just two I was able to remember from top of my head

**Comment by jcansdale** [comment]
@christian-korneck,

> It’s possible using an undocumented api request. See: https://github.com/GeekyEggo/delete-artifact

https://github.com/GeekyEggo/delete-artifact/blob/master/src/runtimeHttpClient.js#L14

Oh, interesting. Where did you come across this API?

**Comment by christian-korneck** [comment]
> Oh, interesting. Where did you come across this API?

@jcansdale That would be a question for @GeekyEggo, I guess. (I'm not involved with his project). Would be curious, too :-)

**Comment by GeekyEggo** [comment]
@jcansdale I discovered the API whilst researching the source for the official GH action that uploads artifacts; more information can be found within https://github.com/GeekyEggo/delete-artifact/blob/master/src/runtimeHttpClient.js

**Comment by jcansdale** [comment]
@GeekyEggo, Thanks for the pointer. I notice the API is marked as `6.0-preview`. Hopefully it will be opened up and documented soon!

**Comment by jacek-jablonski** [comment]
Any news on this? Would love to see `actions/delete-artifact` official action.

**Comment by steve-taylor** [comment]
> Currently there is no experience to remove uploaded artifacts. Is there a scenario you have that you would need this feature for or are you looking to remove an artifact accidentally uploaded?

That's a joke, right?

The scenario is that 30 well paid developers are suddenly and completely blocked because artifacts are unnecessarily persisted beyond the workflow run.

**Comment by princemokut** [comment]
It's almost 12 hours since I deleted artifacts from all previous workflow run and I still get this error: "Create Artifact Container failed: Artifact storage quota has been hit. Unable to upload any new artifacts". This is frustrating our agile process. And I have removed spending limit for about an hour too, still can't run my workflow successfully... :(

**Comment by vHeemstra** [comment]
Using the `actions/github-script` approach can't find any artifacts for a running workflow, and thus can't delete them.

But I found [geekyeggo/delete-artifact@v2](https://github.com/marketplace/actions/delete-artifact) which _does_ remove artifacts created in the same workflow before finishing the workflow.

**Comment by otaviobertucini** [comment]
> It's almost 12 hours since I deleted artifacts from all previous workflow run and I still get this error: "Create Artifact Container failed: Artifact storage quota has been hit. Unable to upload any new artifacts". This is frustrating our agile process. And I have removed spending limit for about an hour too, still can't run my workflow successfully... :(

I manage to stop this error by adding a new step on my deploy script:



      - name: Delete Old Artifacts
        uses: actions/github-script@v6
        id: artifact
        with:
          script: |
            const res = await github.rest.actions.listArtifactsForRepo({
              owner: context.repo.owner,
              repo: context.repo.repo,
            })

            res.data.artifacts
              .forEach(({ id }) => {
                github.rest.actions.deleteArtifact({
                  owner: context.repo.owner,
                  repo: context.repo.repo,
                  artifact_id: id,
                })
              })

Place it right above the upload step and it will delete all previous artifacts of the repo.

**Comment by kolpav** [comment]
I am starting to think that having to

> And I have removed spending limit

is feature not a bug and with my tinfoil hat on I would guess theres number of such people glowing somewhere in GH HQ.

for people landing from google here have this (if solutions from this issue are not working)

```yaml
# https://github.com/actions/upload-artifact/issues/290#issuecomment-1374207010

name: Test

on:
  workflow_run:
    branches: [main]
    workflows: [nonprod,prod,test]
    types:
      - completed
jobs:
  on-success:
    runs-on: ubuntu-latest
    name: Delete artifact
    if: github.event.workflow_run.head_branch == 'main' # feel free to take this out if you don't need it
    steps:
      - id: get-id
        run: | # grab the id of the artifact we just created
          echo "ARTIFACT_ID=$(gh api -H 'Accept: application/vnd.github+json'   ${{github.event.workflow_run.artifacts_url}} --jq .artifacts[].id)" >> $GITHUB_ENV
        env:
          GITHUB_TOKEN: ${{ secrets.GH_REPO_ACCESS_PAT }}
      - run: |
          echo "artifact id is ${{ env.ARTIFACT_ID}}"
          echo "run url is ${{github.event.workflow_run.artifacts_url}}"
      - id: delete-artifact
        run: | # https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28#delete-an-artifact
          gh api --method DELETE -H 'Accept: application/vnd.github+json' /repos/${{github.repository}}/actions/artifacts/${{env.ARTIFACT_ID}}
        env:
          GITHUB_TOKEN: ${{ secrets.GH_REPO_ACCESS_PAT }}
  on-failure:
    runs-on: ubuntu-latest
    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
    steps:
      - run: echo 'The triggering workflow failed'
```

**Comment by DPatrickBoyd** [comment]
> I am starting to think that having to
> 
> > And I have removed spending limit
> 
> is feature not a bug and with my tinfoil hat on I would guess theres number of such people glowing somewhere in GH HQ.
> 
> for people landing from google here have this (if solutions from this issue are not working)
> 
> ```yaml
> # https://github.com/actions/upload-artifact/issues/290#issuecomment-1374207010
> 
> name: Test
> 
> on:
>   workflow_run:
>     branches: [main]
>     workflows: [nonprod,prod,test]
>     types:
>       - completed
> jobs:
>   on-success:
>     runs-on: ubuntu-latest
>     name: Delete artifact
>     if: github.event.workflow_run.head_branch == 'main' # feel free to take this out if you don't need it
>     steps:
>       - id: get-id
>         run: | # grab the id of the artifact we just created
>           echo "ARTIFACT_ID=$(gh api -H 'Accept: application/vnd.github+json'   ${{github.event.workflow_run.artifacts_url}} --jq .artifacts[].id)" >> $GITHUB_ENV
>         env:
>           GITHUB_TOKEN: ${{ secrets.GH_REPO_ACCESS_PAT }}
>       - run: |
>           echo "artifact id is ${{ env.ARTIFACT_ID}}"
>           echo "run url is ${{github.event.workflow_run.artifacts_url}}"
>       - id: delete-artifact
>         run: | # https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28#delete-an-artifact
>           gh api --method DELETE -H 'Accept: application/vnd.github+json' /repos/${{github.repository}}/actions/artifacts/${{env.ARTIFACT_ID}}
>         env:
>           GITHUB_TOKEN: ${{ secrets.GH_REPO_ACCESS_PAT }}
>   on-failure:
>     runs-on: ubuntu-latest
>     if: ${{ github.event.workflow_run.conclusion == 'failure' }}
>     steps:
>       - run: echo 'The triggering workflow failed'
> ```

hey this is my workflow :)

I have a stand alone script on CLI that will delete artifacts based on creation date:

https://gist.github.com/DPatrickBoyd/afb54165df0f51903be3f0edea77f9cb

**Comment by NHAS** [comment]
Why is this issue closed? There still doesnt seem to be a particularly good way to remove artifacts?

**Comment by danijeljw-RPC** [comment]
The issue is not resolved. Continuously closing it is not solving any problems and you've not addressed what people are raising. This has completely stopped my workflow and I need it to resolve to deploy something into prod.

**Comment by relrod** [comment]
+1 for an official `actions/delete-artifact` please.

**Comment by rathorer** [comment]
> > It's almost 12 hours since I deleted artifacts from all previous workflow run and I still get this error: "Create Artifact Container failed: Artifact storage quota has been hit. Unable to upload any new artifacts". This is frustrating our agile process. And I have removed spending limit for about an hour too, still can't run my workflow successfully... :(
> 
> I managed to stop this error by adding a new step on my deploy script:
> 
> ```
>   - name: Delete Old Artifacts
>     uses: actions/github-script@v6
>     id: artifact
>     with:
>       script: |
>         const res = await github.rest.actions.listArtifactsForRepo({
>           owner: context.repo.owner,
>           repo: context.repo.repo,
>         })
> 
>         res.data.artifacts
>           .forEach(({ id }) => {
>             github.rest.actions.deleteArtifact({
>               owner: context.repo.owner,
>               repo: context.repo.repo,
>               artifact_id: id,
>             })
>           })
> ```
> 
> Place it right above the upload step and it will delete all previous artifacts of the repo.

Not sure if this is completely fixed. 
I was facing issue with quota and my pipeline was not running, I could not see any option to delete artifacts from UI, so I used the yml solution to delete all old artifacts. But I am still facing same error.
`Create Artifact Container failed: Artifact storage quota has been hit. Unable to upload any new artifacts`
I also used the artifacts APIs to list artifacts, it shows me 0 artifacts:
```
.github.com/repos/rathorer/proprank/actions/artifacts
{
  "total_count": 0,
  "artifacts": [

  ]
}
```
but the error doesn't go away. Not sure why its keep complaining even if there are no artifacts. 
This is frustrating, I already spent 3-4 days on this.

**Comment by JudgeGazza** [comment]
This seems to work:

https://github.com/marketplace/actions/delete-artifact
